### PR TITLE
Allow user to declare badge block

### PR DESCRIPTION
**Pull request by hadley**
Looks for  and , which  been added to the default templates. 

@maelle could you also please take a look? I decided to mark the start and the end of the badge block in order to make parsing more robust.

### CODE REVIEW

File: R/badge.R [hunk 1]
```diff
@@ -2,7 +2,10 @@
 #'
 #' These helpers produce the markdown text you need in your README to include
 #' badges that report information, such as the CRAN version or test coverage,
-#' and link out to relevant external resources.
+#' and link out to relevant external resources. To add badges auomatically
```
Comment (jennybc) [inline R/badge.R]:
```suggestion
#' and link out to relevant external resources. To add badges automatically
```

File: R/badge.R [hunk 2]
```diff
@@ -2,7 +2,10 @@
...
+#' ensure your badge block is starts with a line containing only
```
Comment (jennybc) [inline R/badge.R]:
```suggestion
#' ensure your badge block starts with a line containing only
```

File: R/badge.R [hunk 3]
```diff
@@ -41,18 +44,49 @@ NULL
...
+    }
+
+    todo(
+      "Add a {field(badge_name)} badge by adding the following line ",
+      "to your README:"
+    )
+    code_block(link)
+    todo(
+      "Allow badges to be added automatically by adding ",
+      "{code('<!-- badges -->')} to the first line of the badges paragraph"
```
Comment (jennybc) [inline R/badge.R]: I think this fell behind the help & code.

File: tests/testthat/test-use-badge.R [hunk 4]
```diff
@@ -21,8 +21,17 @@ test_that("use_binder_badge() the github repository works", {
...
 test_that("use_badge() does nothing if badge seems to pre-exist", {
-  pkg <- scoped_temporary_package()
+  scoped_temporary_package()
   href <- "https://cran.r-project.org/package=foo"
   writeLines(href, proj_path("README.md"))
   expect_false(use_badge("foo", href, "SRC"))
 })
+
+test_that("default readme has placeholders / can add to empty badge block", {
+  withr::local_options(list(usethis.quiet = FALSE))
```
Comment (jennybc) [inline tests/testthat/test-use-badge.R]: This is option is already set globally for tests in `tests/testthat/setup.R`.

File: R/badge.R [hunk 5]
```diff
@@ -41,18 +44,49 @@ NULL
 #' @rdname badges
 #' @export
 use_badge <- function(badge_name, href, src) {
-  if (has_badge(href)) {
+  link <- glue("[![{badge_name}]({src})]({href})")
+  badge_lines <- badge_lines()
```
Comment (jennybc) [inline R/badge.R]: It feels odd to get the line numbers back separate from the README file path and to then re-find the README. But I'm not sure what to do about that.

Comment (hadley) [reply R/badge.R]: I couldn't figure out how to factor it out without making the fallback code (when there's no README, or no badge comments) more complicated. Someone can take another stab at it in #517

**Pull request merged**